posthog/hogql/resolver.py

```diff
@@ -473,6 +484,26 @@ def visit_not(self, node: ast.Not):
         return node
 
     def visit_compare_operation(self, node: ast.CompareOperation):
+        left = self.visit(node.left)
+        left_is_events = self._is_events_table(left)
+        right = self.visit(node.right)
+        right_is_s3 = self._is_s3_cluster(right)
+
+        if (
+            (node.op == ast.CompareOperationOp.In or node.op == ast.CompareOperationOp.NotIn)
+            and left_is_events
+            and right_is_s3
```

Comment:
Would be a tiny tiny bit more efficient to do `self._is_events_table(left)` and `self._is_s3_cluster(right)` here directly, because short-circuiting